pkg/controller/hostoperatorconfig/cache.go

```diff
@@ -0,0 +1,72 @@
+package hostoperatorconfig
+
+import (
+	"context"
+	"sync"
+
+	"github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var configCache = &cache{}
+
+type cache struct {
+	sync.RWMutex
+	config *v1alpha1.HostOperatorConfig
+}
+
+func (c *cache) set(config *v1alpha1.HostOperatorConfig) {
+	c.Lock()
+	defer c.Unlock()
+	c.config = config.DeepCopy()
+}
+
+func (c *cache) get() *v1alpha1.HostOperatorConfig {
+	c.RLock()
+	defer c.RUnlock()
+	return c.config.DeepCopy()
+}
+
+func updateConfig(config *v1alpha1.HostOperatorConfig) {
+	configCache.set(config)
+}
+
+func loadLatest(cl client.Client, namespace string) error {
+	config := &v1alpha1.HostOperatorConfig{}
+	if err := cl.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: "config"}, config); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
```

Comment:
Maybe worth logging such not found error at least?

Reply:
added fd64a4e